var/spack/repos/builtin/packages/flann/package.py

```diff
@@ -0,0 +1,143 @@
+
+    version('1.9.1', '73adef1c7bf8e8b978987e7860926ea6')
+    version('1.8.5', '02a81640b1e9c11796a0413976dc11f5')
+    version('1.8.4', '774b74580e3cbc5b0d45c6ec345a64ae')
+    version('1.8.1', '1f51500e172f5e11fbda05f033858eb6')
+    version('1.8.0', '473150f592c2997e32d5ce31fd3c19a2')
+
+    # Options available in the CMakeLists.txt
+    # Language bindings
+    variant("python",   default=False,
+            description="Build the Python bindings. "
+                        "Module: pyflann. "
+                        "Python2 verified. "
+                        "Python3 claimed, not attainable.")
+    variant("matlab",   default=False, description="Build the Matlab bindings.")
+    # default to true for C because it's a C++ library, nothing extra needed
+    variant("c",        default=True,  description="Build the C bindings.")
+
+    # Must build C bindings for Python / Matlab
+    conflicts("+python", when="~c")
+    conflicts("+matlab", when="~c")
+
+    # Additional options
+    variant("cuda",     default=False, description="Build the CUDA library.")
+    variant("examples", default=False, description="Build the examples.")
+    variant("doc",      default=False, description="Build the documentation.")
+    variant("openmp",   default=True,  description="Use OpenMP multi-threading.")
+    # mpi and hdf5 are the bread and butter of this library, use 'em
+    variant("mpi",      default=True,  description="Use MPI.")
+    variant("hdf5",     default=True,  description="Enable HDF5 support.")
+
+    # Dependencies
+    extends("python",      when="+python")
+    depends_on("py-numpy", when="+python", type=("build", "run"))
+    depends_on("matlab",   when="+matlab")
+    depends_on("cuda",     when="+cuda")
+    depends_on("mpi",      when="+mpi")
+    depends_on("hdf5",     when="+hdf5")
+    # HDF5_IS_PARALLEL actually comes from hdf5+mpi
+    # https://github.com/mariusmuja/flann/blob/06a49513138009d19a1f4e0ace67fbff13270c69/CMakeLists.txt#L108-L112
+    depends_on("boost+mpi+system+serialization+thread", when="+mpi ^hdf5+mpi")
+
+    # Doc deps
+    depends_on("latex", when="+doc")
+
+    # Example uses hdf5.
+    depends_on("hdf5", when="+examples")
+
+    def patch(self):
+        # Fix up the python setup.py call inside the install(CODE
+        filter_file("setup.py install",
+                    '--no-user-cfg setup.py install --prefix=\\"{0}\\"'.format(
+                        self.prefix
+                    ),
+                    "src/python/CMakeLists.txt")
+        # Fix the install location so that spack activate works
+        filter_file("share/flann/python",
+                    site_packages_dir,
+                    "src/python/CMakeLists.txt")
+        # Hack. Don't install setup.py
+        filter_file("install( FILES",
+                    "# install( FILES",
+                    "src/python/CMakeLists.txt", string=True)
+
+    # Tests: require hdf5 and gtest, don't know what to do so ignore...
```

Comment:
For the record, we're working on this. See #1279. I've been adding things like:
```python
# TODO: Add a 'test' deptype
# depends_on('hdf5', type='test')
# depends_on('gtest', type='test')
```
to packages where I noticed there was a test dependency.